docs/modules/ROOT/partials/storage-ceph-cleanup-osd.adoc

```diff
@@ -1,40 +1,54 @@
-. Make a note of the PVC of the old OSD
+. Make a note of the PVCs of the old OSDs
 +
-NOTE: We also extract the name of the PV here, but we'll only delete the PV after removing the node from the cluster.
+NOTE: We also extract the name of the PVs here, but we'll only delete the PVs after removing the nodes from the cluster.
 +
-[source,bash]
-----
-pvc_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
-  "rook-ceph-osd-${OSD_ID}" -ojsonpath='{.metadata.labels.ceph\.rook\.io/pvc}')
-pv_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pvc \
-  "${pvc_name}" -o jsonpath='{.spec.volumeName}')
+[source,bash,subs="attributes+"]
+----
+old_pvc_names=""
+old_pv_names=""
+for node in $(echo -n {osd-replace-list}); do
+  osd_id=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    -l failure-domain="${node}" --no-headers \
+    -o custom-columns="NAME:.metadata.name" | cut -d- -f4)
+
+  pvc_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    "rook-ceph-osd-${osd_id}" -ojsonpath='{.metadata.labels.ceph\.rook\.io/pvc}')
+  pv_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pvc \
+    "${pvc_name}" -o jsonpath='{.spec.volumeName}')
+
+  old_pvc_names="$old_pvc_names $pvc_name"
+  old_pv_names="$old_pv_names $pv_name"
+done
+echo $old_pvc_names
+echo $old_pv_names
 ----
 
-. Check if the OSD deployment needs to be deleted, and delete it if necessary
+. Delete old OSD deployments
 +
-[source,bash]
+[source,bash,subs="attributes+"]
 ----
-kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
-  -l failure-domain="${NODE_TO_REMOVE}"
-# Run this command if the previous command lists a deployment
-kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete deploy \
-  -l failure-domain="${NODE_TO_REMOVE}"
+for node in $(echo -n {osd-replace-list}); do
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete deploy \
+    -l failure-domain="${node}"
+done
 ----
 
-. Clean up PVC and prepare job of the old OSD if necessary
+. Clean up PVCs and prepare jobs of the old OSDs if necessary
```

Comment:
For all of these step descriptions it may make sense to either use the following form

```suggestion
. Clean up PVC(s) and prepare job(s) of the old OSD(s) if necessary
```

Or to have another page attribute which you can use to make the descriptions plural only when necessary

```suggestion
ifdef::multiple-nodes[]
. Clean up PVCs and prepare jobs of the old OSDs if necessary
endif::multiple-nodes[]
ifndef::multiple-nodes[]
. Clean up PVC and prepare job of the old OSD if necessary
endif::multiple-nodes[]
```

Reply:
I do like the first option. The second one just feels like we start to duplicate way to much text again.

Reply:
I was considering suggesting something along the lines of `PVC{plural-suffix}` but that creates a large amount of headache to control the value of `:plural-suffix:`.

Reply:
changed pluralization